Unable to build multiple Vite packages at once

After upgrading to `@emberoider/vite` v1.3.0 from v1.2.1  I’m unable to build multiple packages at once. Got a monorepo with a few packages using the `@ember/addon-blueprint` blueprint. When running their tests in parallel the vite build steps doesn't work after the upgrade.

```
pnpm --filter './packages/**' -r exec vite build
```

```
vite v6.4.0 building for production...
vite v6.4.0 building for production...
vite v6.4.0 building for production...
vite v6.4.0 building for production...
<script src="/testem.js"> in "/tests/index.html" can't be bundled without type="module" attribute
transforming (1) tests/index.html<script src="/testem.js"> in "/tests/index.html" can't be bundled without type="module" attribute
transforming (1) tests/index.html<script src="/testem.js"> in "/tests/index.html" can't be bundled without type="module" attribute
<script src="/testem.js"> in "/tests/index.html" can't be bundled without type="module" attribute
✓ 6 modules transformed.
✓ 6 modules transformed.
✗ Build failed in 304ms
error during build:
[commonjs--resolver] ENOENT: no such file or directory, rename '/private/var/folders/3f/0n0rq49135q31tjykcqvxzrc0000gn/T/embroider-vite-jump/11c9e70b/package.json.tmp' -> '/private/var/folders/3f/0n0rq49135q31tjykcqvxzrc0000gn/T/embroider-vite-jump/11c9e70b/package.json'
    at renameSync (node:fs:1033:11)
    at maybeCaptureNewOptimizedDep (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/@embroider+vite@1.3.0_@embroider+core@4.2.4_@glint+template@1.4.1-unstable.34c4510__@gl_8f88e0346abbd3eb77ea731c030a35cf/node_modules/@embroider/vite/dist/src/resolver.js:214:5)
    at resolveId (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/@embroider+vite@1.3.0_@embroider+core@4.2.4_@glint+template@1.4.1-unstable.34c4510__@gl_8f88e0346abbd3eb77ea731c030a35cf/node_modules/@embroider/vite/dist/src/resolver.js:50:34)
    at async Object.resolveId (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/@embroider+vite@1.3.0_@embroider+core@4.2.4_@glint+template@1.4.1-unstable.34c4510__@gl_8f88e0346abbd3eb77ea731c030a35cf/node_modules/@embroider/vite/dist/src/resolver.js:122:30)
    at async PluginDriver.hookFirstAndGetPlugin (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/rollup@4.49.0/node_modules/rollup/dist/es/shared/node-entry.js:22277:28)
    at async resolveId (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/rollup@4.49.0/node_modules/rollup/dist/es/shared/node-entry.js:20754:26)
    at async ModuleLoader.resolveId (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/rollup@4.49.0/node_modules/rollup/dist/es/shared/node-entry.js:21190:15)
    at async Object.resolveId (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/vite@6.4.0_@types+node@24.3.0_sass@1.53.0_terser@5.43.1/node_modules/vite/dist/node/chunks/dep-N2lvmi1C.js:2835:10)
    at async PluginDriver.hookFirstAndGetPlugin (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/rollup@4.49.0/node_modules/rollup/dist/es/shared/node-entry.js:22277:28)
    at async resolveId (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/rollup@4.49.0/node_modules/rollup/dist/es/shared/node-entry.js:20754:26)
    at async ModuleLoader.resolveId (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/rollup@4.49.0/node_modules/rollup/dist/es/shared/node-entry.js:21190:15)
    at async file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/rollup@4.49.0/node_modules/rollup/dist/es/shared/node-entry.js:21494:50
    at async Promise.all (index 4)
✗ Build failed in 291ms
error during build:
[commonjs--resolver] ENOENT: no such file or directory, rename '/private/var/folders/3f/0n0rq49135q31tjykcqvxzrc0000gn/T/embroider-vite-jump/11c9e70b/package.json.tmp' -> '/private/var/folders/3f/0n0rq49135q31tjykcqvxzrc0000gn/T/embroider-vite-jump/11c9e70b/package.json'
    at renameSync (node:fs:1033:11)
    at maybeCaptureNewOptimizedDep (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/@embroider+vite@1.3.0_@embroider+core@4.2.4_@glint+template@1.4.1-unstable.34c4510__@gl_8f88e0346abbd3eb77ea731c030a35cf/node_modules/@embroider/vite/dist/src/resolver.js:214:5)
    at resolveId (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/@embroider+vite@1.3.0_@embroider+core@4.2.4_@glint+template@1.4.1-unstable.34c4510__@gl_8f88e0346abbd3eb77ea731c030a35cf/node_modules/@embroider/vite/dist/src/resolver.js:50:34)
    at async Object.resolveId (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/@embroider+vite@1.3.0_@embroider+core@4.2.4_@glint+template@1.4.1-unstable.34c4510__@gl_8f88e0346abbd3eb77ea731c030a35cf/node_modules/@embroider/vite/dist/src/resolver.js:122:30)
    at async PluginDriver.hookFirstAndGetPlugin (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/rollup@4.49.0/node_modules/rollup/dist/es/shared/node-entry.js:22277:28)
    at async resolveId (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/rollup@4.49.0/node_modules/rollup/dist/es/shared/node-entry.js:20754:26)
    at async ModuleLoader.resolveId (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/rollup@4.49.0/node_modules/rollup/dist/es/shared/node-entry.js:21190:15)
    at async Object.resolveId (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/vite@6.4.0_@types+node@24.3.0_sass@1.53.0_terser@5.43.1/node_modules/vite/dist/node/chunks/dep-N2lvmi1C.js:2835:10)
    at async PluginDriver.hookFirstAndGetPlugin (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/rollup@4.49.0/node_modules/rollup/dist/es/shared/node-entry.js:22277:28)
    at async resolveId (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/rollup@4.49.0/node_modules/rollup/dist/es/shared/node-entry.js:20754:26)
    at async ModuleLoader.resolveId (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/rollup@4.49.0/node_modules/rollup/dist/es/shared/node-entry.js:21190:15)
    at async file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/rollup@4.49.0/node_modules/rollup/dist/es/shared/node-entry.js:21494:50
    at async Promise.all (index 2)
✓ 5 modules transformed.
/Users/mogstad/Projects/clarify/web-jj/packages/canvas:
 ERR_PNPM_RECURSIVE_EXEC_FIRST_FAIL  Command failed with exit code 1: vite build
✗ Build failed in 326ms
error during build:
[commonjs--resolver] ENOENT: no such file or directory, rename '/private/var/folders/3f/0n0rq49135q31tjykcqvxzrc0000gn/T/embroider-vite-jump/11c9e70b/package.json.tmp' -> '/private/var/folders/3f/0n0rq49135q31tjykcqvxzrc0000gn/T/embroider-vite-jump/11c9e70b/package.json'
    at renameSync (node:fs:1033:11)
    at maybeCaptureNewOptimizedDep (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/@embroider+vite@1.3.0_@embroider+core@4.2.4_@glint+template@1.4.1-unstable.34c4510__@gl_8f88e0346abbd3eb77ea731c030a35cf/node_modules/@embroider/vite/dist/src/resolver.js:214:5)
    at resolveId (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/@embroider+vite@1.3.0_@embroider+core@4.2.4_@glint+template@1.4.1-unstable.34c4510__@gl_8f88e0346abbd3eb77ea731c030a35cf/node_modules/@embroider/vite/dist/src/resolver.js:50:34)
    at async Object.resolveId (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/@embroider+vite@1.3.0_@embroider+core@4.2.4_@glint+template@1.4.1-unstable.34c4510__@gl_8f88e0346abbd3eb77ea731c030a35cf/node_modules/@embroider/vite/dist/src/resolver.js:122:30)
    at async PluginDriver.hookFirstAndGetPlugin (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/rollup@4.49.0/node_modules/rollup/dist/es/shared/node-entry.js:22277:28)
    at async resolveId (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/rollup@4.49.0/node_modules/rollup/dist/es/shared/node-entry.js:20754:26)
    at async ModuleLoader.resolveId (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/rollup@4.49.0/node_modules/rollup/dist/es/shared/node-entry.js:21190:15)
    at async Object.resolveId (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/vite@6.4.0_@types+node@24.3.0_sass@1.53.0_terser@5.43.1/node_modules/vite/dist/node/chunks/dep-N2lvmi1C.js:2835:10)
    at async PluginDriver.hookFirstAndGetPlugin (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/rollup@4.49.0/node_modules/rollup/dist/es/shared/node-entry.js:22277:28)
    at async resolveId (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/rollup@4.49.0/node_modules/rollup/dist/es/shared/node-entry.js:20754:26)
    at async ModuleLoader.resolveId (file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/rollup@4.49.0/node_modules/rollup/dist/es/shared/node-entry.js:21190:15)
    at async file:///Users/mogstad/Projects/clarify/web-jj/node_modules/.pnpm/rollup@4.49.0/node_modules/rollup/dist/es/shared/node-entry.js:21494:50
```

--- a/packages/vite/src/resolver.ts
+++ b/packages/vite/src/resolver.ts
@@ -1,18 +1,19 @@
 import { type Plugin, type ViteDevServer, normalizePath } from 'vite';
-import core, { ModuleRequest, type Resolver } from '@embroider/core';
+import core, { ModuleRequest, type Package, type Resolver } from '@embroider/core';
 const { virtualContent, ResolverLoader, explicitRelative, cleanUrl, tmpdir } = core;
 import { type ResponseMeta, RollupRequestAdapter } from './request.js';
 import { assertNever } from 'assert-never';
 import makeDebug from 'debug';
-import { join, normalize, resolve, dirname } from 'path';
+import { join, normalize, resolve } from 'path';
 import { writeStatus } from './backchannel.js';
 import type { PluginContext, ResolveIdResult } from 'rollup';
 import { externalName } from '@embroider/reverse-exports';
 import fs from 'fs-extra';
 import { createHash } from 'crypto';
 import { BackChannel } from './backchannel.js';
+import { existsSync, mkdtempSync } from 'node:fs';
 
-const { ensureSymlinkSync, writeFileSync, mkdirpSync, renameSync } = fs;
+const { ensureSymlinkSync, writeFileSync, renameSync } = fs;
 
 const debug = makeDebug('embroider:vite');
 
@@ -241,27 +242,21 @@ async function maybeCaptureNewOptimizedDep(
 
   debug('maybeCaptureNewOptimizedDep: doing re-resolve for %s ', foundFile);
 
-  let jumpRoot = join(tmpdir, 'embroider-vite-jump', hashed(pkg.root));
-  let fromFile = join(jumpRoot, 'package.json');
-
-  mkdirpSync(dirname(fromFile));
-  writeFileSync(
-    fromFile + '.tmp',
-    JSON.stringify({
-      name: 'jump-root',
-    }),
-    { flush: true }
-  );
-  renameSync(fromFile + '.tmp', fromFile);
-  try {
-    ensureSymlinkSync(pkg.root, join(jumpRoot, 'node_modules', pkg.name));
-  } catch (error) {
-    //on windows ensureSymlinkSync can throw an EEXIST for invalid reasons
-    if (error.code !== 'EEXIST') {
-      throw error;
+  let jumpRoot = join(tmpdir, 'embroider-vite-jump-' + hashed(pkg.root + '|' + pkg.name));
+  if (!existsSync(jumpRoot)) {
+    let tmp = buildViteJump(pkg);
+    try {
+      renameSync(tmp, jumpRoot);
+    } catch (err) {
+      if (err.code === 'EEXIST') {
+        // somebody raced us and made it first. It's content-addressable so
+        // that's fine.
+      } else {
+        throw err;
+      }
     }
   }
-  let newResult = await context.resolve(target, fromFile);
+  let newResult = await context.resolve(target, join(jumpRoot, 'package.json'));
   if (newResult) {
     if (idFromResult(newResult) === foundFile) {
       // This case is normal. For example, people could be using
@@ -278,4 +273,17 @@ async function maybeCaptureNewOptimizedDep(
   } else {
     return result;
   }
+}
+
+function buildViteJump(pkg: Package) {
+  let tmp = mkdtempSync(join(tmpdir, 'embroider-vite-jump-tmp-'));
+  writeFileSync(
+    join(tmp, 'package.json'),
+    JSON.stringify({
+      name: 'jump-root',
+    }),
+    { flush: true }
+  );
+  ensureSymlinkSync(pkg.root, join(tmp, 'node_modules', pkg.name));
+  return tmp;
 }

